security/pkg/nodeagent/secretfetcher/secretfetcher.go

```diff
@@ -143,8 +163,14 @@ func (sf *SecretFetcher) Init(core corev1.CoreV1Interface) { // nolint:interface
 			return core.Secrets(namespace).Watch(options)
 		},
 	}
+
+	resyncPeriod := 0 * time.Second
+	if env, err := time.ParseDuration(os.Getenv(secretControllerResyncPeriod)); err == nil {
+		resyncPeriod = env
+	}
+
 	sf.scrtStore, sf.scrtController =
-		cache.NewInformer(scrtLW, &v1.Secret{}, secretResyncPeriod, cache.ResourceEventHandlerFuncs{
+		cache.NewInformer(scrtLW, &v1.Secret{}, resyncPeriod, cache.ResourceEventHandlerFuncs{
```

Comment:
We need to wait for synced - or use SharedInformer if this is not possible with Informer

Reply:
while ! sf.scrtController.HasSynced() { sleep(1) }

Reply:
The library already provides a method cache.WaitForCacheSync(), I add a call to WaitForCacheSync() in func (sf *SecretFetcher) Run(ch chan struct{}) 
Thanks!